planner/cascades/transformation_rules.go

```diff
@@ -1286,3 +1289,46 @@ func (r *MergeAdjacentSelection) OnTransform(old *memo.ExprIter) (newExprs []*me
 	newSelExpr.SetChildren(childGroups...)
 	return []*memo.GroupExpr{newSelExpr}, true, false, nil
 }
+
+// MergeAdjacentLimit merge the adjacent limit.
+type MergeAdjacentLimit struct {
+	baseRule
+}
+
+// NewRuleMergeAdjacentLimit creates a new Transformation MergeAdjacentLimit.
+// The pattern of this rule is `Limit->Limit->X`.
+func NewRuleMergeAdjacentLimit() Transformation {
+	rule := &MergeAdjacentLimit{}
+	rule.pattern = memo.BuildPattern(
+		memo.OperandLimit,
+		memo.EngineAll,
+		memo.NewPattern(memo.OperandLimit, memo.EngineAll),
+	)
+	return rule
+}
+
+// OnTransform implements Transformation interface.
+// This rule tries to merge adjacent limit.
+func (r *MergeAdjacentLimit) OnTransform(old *memo.ExprIter) (newExprs []*memo.GroupExpr, eraseOld bool, eraseAll bool, err error) {
+	limit := old.GetExpr().ExprNode.(*plannercore.LogicalLimit)
+	child := old.Children[0].GetExpr().ExprNode.(*plannercore.LogicalLimit)
+	childGroups := old.Children[0].GetExpr().Children
+
+	if child.Count <= limit.Offset {
+		tableDual := plannercore.LogicalTableDual{RowCount: 0}.Init(child.SCtx(), child.SelectBlockOffset())
+		tableDual.SetSchema(child.Schema())
+		tableDual.SetOutputNames(child.OutputNames())
```

Comment:
This may cause a `nil pointer bug`.  In cascades planner, we didn't set the `children` of `LogcalPlan`, but `child.Schema()` may use `child.children[0].Schema()`. 
Use `tableDual.SetSchema(old.GetExpr().Schema())` instead.
As for `OutputNames`, is it nescessary to set? I think we can just remove it here.